include/dd/Package.hpp

```diff
@@ -676,6 +679,201 @@ namespace dd {
+                double prob = probsMone[ptr];
+
+                if (!ptr->e.at(0).w.approximatelyZero()) {
+                    const fp tmp1 = prob * ComplexNumbers::mag2(ptr->e.at(0).w);
+
+                    if (visited_nodes2.find(ptr->e.at(0).p) != visited_nodes2.end()) {
+                        probsMone[ptr->e.at(0).p] = probsMone[ptr->e.at(0).p] + tmp1;
+                    } else {
+                        probsMone[ptr->e.at(0).p] = tmp1;
+                        visited_nodes2.insert(ptr->e.at(0).p);
+                        q.push(ptr->e.at(0).p);
+                    }
+                }
+
+                if (!ptr->e.at(1).w.approximatelyZero()) {
+                    const fp tmp1 = prob * ComplexNumbers::mag2(ptr->e.at(1).w);
+
+                    if (visited_nodes2.find(ptr->e.at(1).p) != visited_nodes2.end()) {
+                        probsMone[ptr->e.at(1).p] = probsMone[ptr->e.at(1).p] + tmp1;
+                    } else {
+                        probsMone[ptr->e.at(1).p] = tmp1;
+                        visited_nodes2.insert(ptr->e.at(1).p);
+                        q.push(ptr->e.at(1).p);
+                    }
+                }
+            }
+
+            fp pzero{0}, pone{0};
+
+            if (assume_probability_normalization) {
+                while (!q.empty()) {
+                    vNode* ptr = q.front();
+                    q.pop();
+
+                    if (!ptr->e.at(0).w.approximatelyZero()) {
+                        pzero += probsMone[ptr] * ComplexNumbers::mag2(ptr->e.at(0).w);
+                    }
+
+                    if (!ptr->e.at(1).w.approximatelyZero()) {
+                        pone += probsMone[ptr] * ComplexNumbers::mag2(ptr->e.at(1).w);
+                    }
+                }
+            } else {
+                std::unordered_map<vNode*, double> probs;
+                assignProbabilities(root_edge, probs);
+
+                while (!q.empty()) {
+                    vNode* ptr = q.front();
+                    q.pop();
+
+                    if (!ptr->e.at(0).w.approximatelyZero()) {
+                        pzero += probsMone[ptr] * probs[ptr->e.at(0).p] * ComplexNumbers::mag2(ptr->e.at(0).w);
+                    }
+
+                    if (!ptr->e.at(1).w.approximatelyZero()) {
+                        pone += probsMone[ptr] * probs[ptr->e.at(1).p] * ComplexNumbers::mag2(ptr->e.at(1).w);
+                    }
+                }
+            }
+
+            if (std::abs(pzero + pone - 1) > epsilon) {
+                throw std::runtime_error("Numerical instability occurred during measurement: |alpha|^2 + |beta|^2 = " + std::to_string(pzero) + " + " + std::to_string(pone) + " = " +
+                                         std::to_string(pzero + pone) + ", but should be 1!");
+            }
+
+            const fp sum = pzero + pone;
+
+            GateMatrix measure_m{
+                    complex_zero, complex_zero,
+                    complex_zero, complex_zero};
```

Comment:
```suggestion
            GateMatrix measurementMatrix{
                    complex_zero, complex_zero,
                    complex_zero, complex_zero};
```
Should `complex_zero` maybe also be called `complexZero`?